modules/csg/csg_shape.cpp

```diff
@@ -142,6 +142,7 @@ bool CSGShape3D::is_root_shape() const {
 
 void CSGShape3D::set_snap(float p_snap) {
 	snap = p_snap;
```

Comment:
Prevent `_make_dirty()` from being called if the value hasn't actually changed:

```suggestion
void CSGShape3D::set_snap(float p_snap) {
	if (p_snap == snap) {
		return;
	}
	
	snap = p_snap;
```